common/ts/publish-task.ts

```diff
@@ -22,28 +22,36 @@ export default async function publishTask(endpointType: EndpointType) {
   const endpoint = new Endpoint(endpointData.type, endpointData.data);
   const metrics = await Metrics.getAllMetrics(endpoint);
 
-  const taskReport = await TaskReport.createTaskReportFromFile();
   const timeoutSec = timeoutInSeconds();
-  try {
-    const task = await Task.waitForTaskCompletion(endpoint, taskReport.ceTaskId, timeoutSec);
-    const analysis = await Analysis.getAnalysis(
-      task.analysisId,
-      endpoint,
-      metrics,
-      taskReport.dashboardUrl
-    );
-    publishBuildSummary(analysis.getHtmlAnalysisReport(), endpoint.type);
-  } catch (e) {
-    if (e instanceof TimeOutReachedError) {
-      tl.warning(
-        `Task '${
-          taskReport.ceTaskId
-        }' takes too long to complete. Stopping after ${timeoutSec}s of polling. No quality gate will be displayed on build result.`
-      );
-    } else {
-      throw e;
-    }
-  }
+  const buildSummariesPromises = TaskReport.createTaskReportsFromFiles().map((promise) => {
+    return promise.then(async (taskReport) => {
+      try {
+        const task = await Task.waitForTaskCompletion(endpoint, taskReport.ceTaskId, timeoutSec);
+        const analysis = await Analysis.getAnalysis(
+          task.analysisId,
+          task.componentName,
+          endpoint,
+          metrics,
+          taskReport.dashboardUrl
+        );
+        return analysis.getHtmlAnalysisReport();
+      } catch (e) {
+        if (e instanceof TimeOutReachedError) {
+          tl.warning(
+            `Task '${
+            taskReport.ceTaskId
+            }' takes too long to complete. Stopping after ${timeoutSec}s of polling. No quality gate will be displayed on build result.`
+          );
+        } else {
+          throw e;
+        }
+      }
+    });
+  });
+
+  const buildSummaries = await Promise.all(buildSummariesPromises);
+  const reducer = (accumulator: string, currentValue: string) => `${accumulator}\r\n${currentValue}`;
+  publishBuildSummary(buildSummaries.reduce(reducer), endpoint.type);
```

Comment:
And with the previous change, you don't need that reduce function and you can just use `Array.join` on the analyses:
```ts
publishBuildSummary(analyses.join('\r\n'), endpoint.type);
```